flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,434 @@
+from flexget.utils.search import normalize_unicode
+from sqlalchemy import Column, Unicode, String, DateTime
+
+
+DETECT_2FA = ("Authenticator Code", "TOTP code")
+log = logging.getLogger('gazelle')
+Base = db_schema.versioned_base('gazelle_session', 0)
+
+
+class GazelleSession(Base):
+    __tablename__ = 'gazelle_session'
+
+    username = Column(Unicode, primary_key=True)
+    base_url = Column(String, primary_key=True)
+
+    authkey = Column(String)
+    passkey = Column(String)
+    _cookies = Column('cookie', Unicode)
+    cookies = json_synonym('_cookies')
+    expires = Column(DateTime)
+
+
+class InputGazelle(object):
+    """A plugin that searches a Gazelle-based website"""
+
+    # Aliases for config -> api params
+    ALIASES = {
+        "artist": "artistname",
+        "album": "groupname",
+        "leech_type": "freetorrent",
+        "release_type": "releasetype",
+        "tags": "taglist",
+        "tag_type": "tags_type",
+        "search": "searchstr",
+        "log": "haslog",
+    }
+
+    # API parameters
+    # None means a raw value entry (no validation other than schema)
+    # A dict means an enum with a config -> api mapping
+    # A list is an enum with no mapping
+    PARAMS = {
+        "searchstr": None
+    }
+
+    def __init__(self):
+        self.base_url = None
+
+    def _key(self, key):
+        """Gets the API key name from the entered key"""
+        if key in self.ALIASES:
+            return self.ALIASES[key]
+        return key
+
+    def _opts(self, key):
+        """Gets the options for the specified key"""
+        return self.PARAMS[self._key(key)]
+
+    def _getval(self, key, val):
+        """Gets the value for the specified key based on a config option"""
+        opts = self._opts(key)
+        if isinstance(opts, dict):
+            # Translate the input value to the API value
+            # The str cast converts bools to 'True'/'False' for use as keys
+            # This allows for options that have True/False/Other values
+            return opts[str(val)]
+        elif isinstance(val, list):
+            # Fix yaml parser making a list out of a string
+            return ",".join(val)
+        return val
+
+    def params_from_config(self, config):
+        """Filter params and map config values -> api values"""
+        ret = {}
+        for k, v in config.items():
+            key = self._key(k)
+            if key in self.PARAMS:
+                ret[key] = self._getval(k, v)
+        return ret
+
+    def setup(self, task, config):
+        """Set up a session and log in"""
+        self._session = Session()
+        base_url = config.get('base_url', "").rstrip("/")
+        if base_url:
+            if self.base_url and self.base_url != base_url:
+                log.warning("Using plugin designed for %s on %s - "
+                            "things may break", self.base_url, base_url)
+            self.base_url = base_url
+
+        if not self.base_url:
```

Comment:
could replace this to `else`

Reply:
Using `else` would break in the case where no `base_url` is in the config but one is set up on the subclass so it doesn't need to be.